src/program/gc/gc.lua

```diff
@@ -18,10 +19,14 @@ function run (args)
 
    -- Unlink stale snabb resources.
    for _, pid in ipairs(shm.children("//")) do
-      if not syscall.kill(tonumber(pid), 0) then
+      if tonumber(pid) and not syscall.kill(tonumber(pid), 0) then
          shm.unlink("//"..pid)
       end
    end
+
+   -- unmap hugepages
+   memory.cleanup()
```

Comment:
The `memory.cleanup()` marks for deletion all the hugepages registered in the `map_ids` structure, but that only means they will be deleted when the last process to use a given hugepage unmaps it (or finishes).  While a hugepage is used by a process, it can be used by another process.  I tried to make it fail by marking at several times, but it always worked.

I'm even tempted to restore the `shmctl()` call just after allocation, as it was before, making this `cleanup()` function unnecessary.  I guess it would only fail if:

- process A allocates some packets from its pool
- those packets are transferred to process B
- process A finishes
- since no other process has mapped A pool, it's destroyed
- process B receives the packet pointers and tries to map it.

It not only seems very unlikely, but also useless.... can anybody come up with a different scenario that could fail because of marking pages as soon as allocated?